src/main/java/Duke.java

```diff
@@ -1,10 +1,180 @@
-public class Duke {
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.stage.Stage;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+/**
+ * Represent the main class to run the Duke program.
+ */
+public class Duke extends Application {
+    private Ui ui;
+    private Storage storage;
+    private Parser parser;
+    private TaskList tasks;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+    
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.parser = new Parser();
+        this.tasks = new TaskList(storage.load());
+    }
+    
+    public Duke() {
+        this.ui = new Ui();
+        this.storage = new Storage("data/storage/duke.txt");
+        this.parser = new Parser();
+        this.tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Start the Duke program.
+     */
+    public void run() {
+        String input = "";
+        String output;
+        ui.showWelcome();
+
+        while (!input.equals("bye")) {
+            input = ui.readInput();
+
+            try {
+                output = parser.parse(input, tasks, storage);
+            } catch (DukeException e) {
+                output = e.getMessage();
+            }
+
+            ui.showOutput(output);
+        }
+    }
+
+    /**
+     * The main method for Duke class.
+     * @param args unused
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke("data/storage/duke.txt");
+        duke.run();
+    }
+
+    @Override
+    public void start(Stage stage) {
```

Comment:
Would be a good idea to consider refactoring this into a separate class to handle the UI of JavaFx instead of having it all in the Duke class. If we try adhere to the SRP, it would probably be best for Duke to handle the underlying implementation logic of interacting with commands? Then the separate class could handle the UI?